src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -230,6 +231,7 @@ public Task<TElement> DequeueAsync(CancellationToken cancellationToken = default
         /// Note: The early cancellation behavior is intentional.
         /// </summary>
         [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/23582", OftenCompletesSynchronously = true)]
+        [SuppressMessage("Usage", "VSTHRD003:Avoid awaiting foreign Tasks", Justification = "This implementation matches the behavior of ThreadingTools.WithCancellation.")]
         private static Task<T> WithCancellation<T>(Task<T> task, CancellationToken cancellationToken)
```

Comment:
There is no `await` here. Not sure what we are fixing for this rule.

Reply:
This boils down to the same situation as https://github.com/dotnet/roslyn/pull/34342#discussion_r267991927. However, in this case we know the behavior is correct because I updated it to precisely follow the vs-threading implementation of `WithCancellation<T>`. The only remaining question is whether the analyzers in vs-threading somehow special-case this method.